src/gf.c

```diff
@@ -2701,7 +2703,7 @@ jl_code_instance_t *jl_compile_method_internal(jl_method_instance_t *mi, size_t
             // Something went wrong. Bail to the fallback path.
             codeinst = NULL;
         }
-        else if (did_compile) {
+        else if (did_compile && codeinst->owner == jl_nothing)
```

Comment:
@vchuravy I think this is the only place where we might we get a non-nothing-owner CI, but please do check my work